qiskit_ionq/exceptions.py

```diff
@@ -168,18 +168,12 @@ def __str__(self):
             f"error_type={self.error_type!r})"
         )
 
-    def __getstate__(self) -> object:
+    def __reduce__(self):
```

Comment:
do we understand why __reduce__ works? __get__ and __set__ are pretty standard way for serialization?

Reply:
i just tested this out and it seems like `__getstate__` and `__setstate__` are relying on an instantiated object (which is why we see the error `TypeError: IonQAPIError.__init__() missing 4 required positional arguments: 'status_code', 'headers', 'body', and 'error_type'`) but `__reduce__` returns the class and arguments to instantiate it.